### DISCUSSION TITLE
useMutationState

### DISCUSSION THREAD

**Discussion by Alexandredc** [TanStack/query#3033] [Ideas] [answered]
Hello guys,

I double check for this feature and I didn't find any suggestion about it.
It would be awesome to have a `useMutateState` function like `useQueryState`.

**Here is my use case : **
In my application, I can mutate a ressource asynchrously and change the route before sur mutation is completed.
Now, imagine the user come back to the previous route. I would like to know if there is an ongoing mutation to indicate to the user that the last mutation is not completed yet.

Maybe I missed an existing solution for that use case ?

Thanks!

**Comment by TkDodo** [comment, marked as answer]
useIsMutating should cover this case.

**Reply by Alexandredc** [reply]
Thanks for your quick reply @TkDodo 
This fit well with my use case.
